### PR TITLE
[Sessions] chore: Use depinject for sessions mgr construction

### PR DESCRIPTION
**Pull request by red-0ne**
### Human Summary

Use depinject to construct `relayerSessionsManager`.

### Summary generated by Reviewpad on 09 Nov 23 23:19 UTC
This pull request introduces several changes. 

The first patch (1/3) adds a new file `options.go` to the `pkg/relayer/session` package, which contains a new function `WithStoresDirectory`. This function sets the path on disk where KVStore data files are created. 

The second patch (2/3) adds a small improvement to the `WithStoresDirectory` function. It changes the error message when the session tree store path already exists. 

The third patch (3/3) improves the readability of the error messages in the `errors.go` file. It updates the error codespace and the error messages to make them more descriptive. It also updates the error message when the session tree store path already exists.

Overall, these changes aim to enhance the management of sessions and their corresponding data files in the `relayer` package.

## Type of change

Select one or more:

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Verify Localnet manually**: See the instructions [here](TODO: add link to instructions)

## Sanity Checklist

- [ ] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: pkg/relayer/session/errors.go [hunk 1]
```diff
@@ -8,4 +8,5 @@ var (
 	ErrSessionTreeNotClosed         = sdkerrors.Register(codespace, 2, "session tree not closed")
 	ErrSessionStorePathExists       = sdkerrors.Register(codespace, 3, "session store path already exists")
 	ErrSessionTreeProofPathMismatch = sdkerrors.Register(codespace, 4, "session tree proof path mismatch")
+	ErrUndefinedStoresDirectory     = sdkerrors.Register(codespace, 5, "undefined stores directory")
```
Comment (Olshansk) [inline pkg/relayer/session/errors.go:11]: This goes a long way when we hit real issues

```suggestion
	ErrSessionTreeUndefinedStoresDirectory      = sdkerrors.Register(codespace, 5, "directory undefined for where to key-value stores should be saved on disk")
```

File: pkg/relayer/session/session.go [hunk 2]
```diff
@@ -136,3 +147,12 @@ func (rs *relayerSessionsManager) removeFromRelayerSessions(sessionHeader *sessi
 		delete(rs.sessionsTrees, sessionHeader.SessionEndBlockHeight)
 	}
 }
+
+// validateConfig validates the relayerSessionsManager's configuration.
+func (rp *relayerSessionsManager) validateConfig() error {
```
Comment (Olshansk) [inline pkg/relayer/session/session.go:152]:
```suggestion
// TODO_TEST: Add unit tests to validate these configurations
func (rp *relayerSessionsManager) validateConfig() error {
```

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: pkg/relayer/session/options.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package session
+
+import (
+	"github.com/pokt-network/poktroll/pkg/relayer"
+)
+
+// WithStoresDirectory sets the path on disk where KVStore data files are created.
```
Comment (bryanchriswhite) [inline pkg/relayer/session/options.go:7]: Is it worth trying to articulate what those files entail (e.g. session SMSTs, etc.)?

**Review by bryanchriswhite — APPROVED** [review]
Left a couple of comments but otherwise G2G! :rocket: 

Preemptively approving.

**Review by Olshansk — APPROVED** [review]

**Pull request merged**